apps/web/src/pages/activities/services/chart-bar/data.service.ts

```diff
@@ -0,0 +1,71 @@
+import * as cloneDeep from 'lodash.clonedeep';
+import moment from 'moment';
+import { ScriptableContext } from 'chart.js';
+import { IActivityGraphStats, IChartData } from '../../interfaces';
+import { activityGraphStatsMock } from '../../consts';
+import { colors } from '../../../../design-system';
+
+export function getChartData(data: IActivityGraphStats[] | undefined, isDark: boolean): IChartData {
+  if (!data || data?.length === 0) {
+    return buildChartDataContainer(activityGraphStatsMock, isDark);
+  }
+
+  if (data.length < 7) {
+    return buildChartDataContainer(fillWeekData(data), isDark);
+  }
+
+  return buildChartDataContainer(data, isDark);
+}
+
+function buildChartDataContainer(data: IActivityGraphStats[], isDark: boolean): IChartData {
+  return {
+    labels: buildChartDateLabels(data),
+    datasets: [
+      {
+        backgroundColor: isDark ? colors.B20 : colors.BGLight,
+        hoverBackgroundColor: createGradientColor(),
+        data: buildChartData(data),
+        borderRadius: 7,
+      },
+    ],
+  };
+}
+
+function fillWeekData(data: IActivityGraphStats[]) {
+  const fullWeekData = cloneDeep(data);
+  // eslint-disable-next-line no-plusplus
+  for (let i = data.length - 1; i < 6; i++) {
+    const earliestDate = fullWeekData[i]._id;
+    const newDate = moment(earliestDate).subtract(1, 'days').format('YYYY-MM-DD');
```

Comment:
```suggestion
    const newDate = moment(earliestDate, 'YYYY-MM-DD').subtract(1, 'days').format('YYYY-MM-DD');
```
moment accepts a second param as the structure of the data, this might eliminate the bug we had.